packages/components/src/base-control/index.js

```diff
@@ -3,14 +3,21 @@
  */
 import classnames from 'classnames';
 
+/**
+ * Internal dependencies
+ */
+import VisuallyHidden from '../visually-hidden';
+
 function BaseControl( { id, label, hideLabelFromVision, help, className, children } ) {
 	return (
 		<div className={ classnames( 'components-base-control', className ) }>
 			<div className="components-base-control__field">
-				{ label && id && <label
-					className={ classnames( 'components-base-control__label', { 'screen-reader-text': hideLabelFromVision } ) }
-					htmlFor={ id }>{ label }
-				</label> }
+				{ label && id && hideLabelFromVision && <VisuallyHidden
+					as="label"
+					htmlFor={ id }>{ label }</VisuallyHidden> }
```

Comment:
I'm wondering if there is any reason to be concerned with losing the `components-base-control__label` class on this label? Maybe it's fine, but wondering if it would be better to set up VisuallyHidden to pass through any additional class names and just replace screen-reader-text?

Reply:
Great noticing, I did consider this (for awhile) and I came to the conclusion that it is better to remove it.

First, having the extra class does not provide any semantic knowledge, the class name isn't used for this and it is already a label tag, so we don't need more info here.

Second, it is probably better to remove because all the CSS properties we want set are the ones to used to hide it visually, any additional attributes set via another class may break the visually hiding. Likely unintentional because using the same class for both visually shown and not shown would be unexpected.

Reply:
Cool, just thought I would call it out to make sure, that makes sense to me